### DISCUSSION TITLE
Type of Payloads Sliver Can generate

### DISCUSSION THREAD

**Discussion by arishwani** [BishopFox/sliver#755] [Q&A] [answered]
Hi Everyone, 

What type of payload Sliver supports other than ".exe" and what's the syntax to generate it? 

Thank you

**Comment by rkervella** [comment, marked as answer]
From `help generate`:

```
  -f, --format             string    Specifies the output formats, valid values are: 'exe', 'shared' (for dynamic libraries), 'service' (see `psexec` for more info) and 'shellcode' (windows only) (default: exe)
```

**Comment by arishwani** [comment]
@rkervella how would generate "dll", "python", "js", etc., I tried that format and didn't work for me

**Reply by rkervella** [reply]
As the help string specifies, we only support:

- `exe` : typical executable format, all platforms
- `shared`: shared library (Windows/MacOS/Linux)
- `shellcode`: only supported for Windows platform
- `service`: Specific executable format, Windows only

**Reply by moloch--** [reply]
`shared` = `dll` when using `--os windows`

**Reply by arishwani** [reply]
Thanks a lot that worked great, I appreciate the reply.
